plugin/pkg/admission/nodeenv/admission.go

```diff
@@ -0,0 +1,234 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nodeenv
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"reflect"
+
+	"github.com/golang/glog"
+
+	"k8s.io/kubernetes/pkg/admission"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/client/cache"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util/yaml"
+	"k8s.io/kubernetes/pkg/watch"
+)
+
+func init() {
+	admission.RegisterPlugin("PodNodeEnvironment", func(client clientset.Interface, config io.Reader) (admission.Interface, error) {
+		pluginConfig := readConfig(config)
+		plugin := NewPodNodeEnvironment(client, pluginConfig.PodNodeEnvironmentPluginConfig)
+		plugin.reflector.Run()
+		return plugin, nil
+	})
+}
+
+// podNodeEnvironment is an implementation of admission.Interface.
+type podNodeEnvironment struct {
+	*admission.Handler
+	client    clientset.Interface
+	store     cache.Store
+	reflector *cache.Reflector
+	// global default tolerations in a cluster, If a namespace is
+	// not assigned tolerations, it gets this by default.
+	clusterDefaultTolerations []api.Toleration
+}
+
+type pluginConfig struct {
+	PodNodeEnvironmentPluginConfig []api.Toleration
+}
+
+// readConfig reads default value of clusterDefaultTolerations
+// from the file provided with --admission-control-config-file
+// If the file is not supplied, it defaults to ""
+// The format in a file:
+// podNodeEnvironmentPluginConfig:
+//  clusterDefaultTolerations: <tolerations>
```

Comment:
see comment above about the config. Seems like the comment matches what I suggested.